File: refactor.js
const fs = require('fs');
let code = fs.readFileSync('src/App.jsx', 'utf8');

// 1. Extract and remove Event Mode Badge
const eventModeStart = code.indexOf('{/* Event Mode Badge/Toggle Switch */}');
const eventModeEnd = code.indexOf('})()}', eventModeStart) + 5;
const eventModeCode = code.substring(eventModeStart, eventModeEnd);
code = code.substring(0, eventModeStart) + code.substring(eventModeEnd);

// 2. Extract and remove Simple Mode Toggle
const simpleModeStart = code.indexOf('{/* Simple Mode Toggle */}');
const simpleModeEnd = code.indexOf('</button>', simpleModeStart) + 9;
const simpleModeCode = code.substring(simpleModeStart, simpleModeEnd);
code = code.substring(0, simpleModeStart) + code.substring(simpleModeEnd);

// 3. Extract and remove Dark mode toggle
const darkModeStart = code.indexOf('{/* Dark mode toggle */}');
const darkModeEnd = code.indexOf('</button>', darkModeStart) + 9;
const darkModeCode = code.substring(darkModeStart, darkModeEnd);
code = code.substring(0, darkModeStart) + code.substring(darkModeEnd);

// 4. Remove User Info and Logout button from header
const userInfoStart = code.indexOf('<div className="header-user-info"');
const userInfoEnd = code.indexOf('</button>', code.indexOf('onClick={handleLogout}', userInfoStart)) + 9;
code = code.substring(0, userInfoStart) + code.substring(userInfoEnd);

// 5. Inject them into MoreDrawer
const moreDrawerTarget = "<div style={{ display: 'flex', flexDirection: 'column', gap: '8px' }}>";
const injection = \
              {/* Quick Settings */}
              <div style={{ display: 'flex', gap: '8px', flexWrap: 'wrap', marginBottom: '16px' }}>
                \
                \
                \
              </div>
\;
code = code.replace(moreDrawerTarget, moreDrawerTarget + '\\n' + injection);

fs.writeFileSync('src/App.jsx', code);

File: refactor.py
import os

with open('src/App.jsx', 'r', encoding='utf-8') as f:
    code = f.read()

# 1. Extract Event Mode
start = code.find('{/* Event Mode Badge/Toggle Switch */}')
end = code.find('})()}', start) + 5
event_mode_code = code[start:end]
code = code[:start] + code[end:]

# 2. Extract Simple UI
start = code.find('{/* Simple Mode Toggle */}')
end = code.find('</button>', start) + 9
simple_ui_code = code[start:end]
code = code[:start] + code[end:]

# 3. Extract Dark Mode
start = code.find('{/* Dark mode toggle */}')
end = code.find('</button>', start) + 9
dark_mode_code = code[start:end]
code = code[:start] + code[end:]

# 4. Extract and remove User Info + Logout
start = code.find('<div className="header-user-info"')
end = code.find('</button>', code.find('onClick={handleLogout}', start)) + 9
user_info_code = code[start:end]
code = code[:start] + code[end:]

# 5. Inject into MoreDrawer
target = "<div style={{ display: 'flex', flexDirection: 'column', gap: '8px' }}>"

injection = f'''              {{/* Quick Settings */}}
              <div style={{{{ display: 'flex', gap: '12px', flexWrap: 'wrap', marginBottom: '16px', background: 'rgba(255,255,255,0.03)', padding: '16px', borderRadius: '16px', border: '1px solid rgba(255,255,255,0.05)', alignItems: 'center', justifyContent: 'center' }}}}>
                {event_mode_code.replace("alignSelf: 'center'", "")}
                {simple_ui_code}
                {dark_mode_code}
              </div>'''

code = code.replace(target, target + '\n' + injection)

with open('src/App.jsx', 'w', encoding='utf-8') as f:
    f.write(code)

print("Done")
